File: nigeria_compliance_client/public/js/customer.js
frappe.ui.form.on('Customer', {
    refresh(frm) {
        // your code here
    },
    custom_b2b_customer(frm) {
        if (frm.doc.custom_b2b_customer == 1) {
            if (frm.doc.custom_b2b_customer == 1 && frm.doc.custom_b2c_customer == 1) {
                frm.set_value("custom_b2c_customer", 0)
            }
        }
    },
    custom_b2c_customer(frm) {
        if (frm.doc.custom_b2c_customer == 1) {
            if (frm.doc.custom_b2c_customer && frm.doc.custom_b2b_customer == 1) {
                frm.set_value("custom_b2b_customer", 0)
            }
        }
    }
})

File: nigeria_compliance_client/nigeria_compliance_client/hooks/customer.py
import frappe

def validate(doc, method=None):
    if doc.custom_b2b_customer and doc.custom_b2c_customer:
        frappe.throw("B2B Customer and B2C Customer cannot be same")

    if doc.custom_b2b_customer:
        check_address_for_b2b_customer(doc)

def check_address_for_b2b_customer(doc):
    if not frappe.db.exists("Dynamic Link", {"link_doctype": "Customer", "link_name": doc.name, "parenttype": "Address"}):
        frappe.throw("Address is required for B2B Customer")

